File: backend/app/routes/compliance.py
"""
GET  /api/compliance/documents — list loaded compliance docs
POST /api/compliance/query     — direct compliance question (no data agent)
"""
from fastapi import APIRouter, HTTPException, Request
from pydantic import BaseModel

router = APIRouter()


class ComplianceQueryRequest(BaseModel):
    question: str
    session_id: str | None = None


@router.get("/compliance/documents")
async def list_compliance_documents(request: Request):
    from app.core.compliance_kb import get_compliance_kb
    kb = get_compliance_kb()
    if not kb.is_loaded:
        return {"documents": [], "loaded": False}
    return {"documents": kb.list_documents(), "loaded": True, "total_chunks": len(kb.chunks)}


@router.post("/compliance/query")
async def compliance_query(request: Request, body: ComplianceQueryRequest):
    from app.agents.compliance_agent import answer_compliance_question
    try:
        result = await answer_compliance_question(body.question, schema=None)
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Compliance query failed: {str(e)}")

File: backend-node/src/routes/compliance.ts
/**
 * GET  /api/compliance/documents
 * POST /api/compliance/query
 * Mirrors Python backend/app/routes/compliance.py.
 */
import { Router, Request, Response } from 'express';
import { getComplianceKb } from '../core/complianceKb';
import { answerComplianceQuestion } from '../agents/complianceAgent';

const router = Router();

router.get('/compliance/documents', (_req: Request, res: Response): void => {
  const kb = getComplianceKb();
  if (!kb.isLoaded) {
    res.json({ documents: [], loaded: false });
    return;
  }
  res.json({ documents: kb.listDocuments(), loaded: true, total_chunks: kb.chunkCount });
});

router.post('/compliance/query', async (req: Request, res: Response): Promise<void> => {
  try {
    const { question } = req.body;
    if (!question) {
      res.status(400).json({ detail: 'question is required.' });
      return;
    }
    const result = await answerComplianceQuestion(question, undefined);
    res.json(result);
  } catch (e: any) {
    res.status(500).json({ detail: `Compliance query failed: ${e.message}` });
  }
});

export default router;
